### PR TITLE
Disable GPL code and Add RISC-V support via Capstone 5

### PR DESCRIPTION
**Pull request by wargio**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

1. Extends the support of `use_gpl` meson option to rizin.

When `-Duse_gpl=false` the following plugins are disabled:
```
Asm: arc, cris_gnu, hppa_gnu, lanai_gnu, mips_gnu, nios2, riscv, sh, sparc_gnu, tricore, vax, xtensa, z80
Analysis: arc, cris, mips_gnu, nios2, riscv, sh, sparc_gnu, tricore, vax, xtensa, z80
Debug: qnx
IO: qnx
```
Also `sflib` files are not installed.

2. Adds RISC-V support via Capstone 5

When Capstone 5 is detected, adds the analysis and asm plugins as `riscv_cs` to rizin. 

3. Adds new CI workflow to build rizin with `-Duse_gpl=false` when `gpl` keyword is found or when is not `push_request`

### CODE REVIEW

File: librz/analysis/meson.build [hunk 1]
```diff
@@ -165,6 +148,31 @@ rz_analysis_sources = [
...
+endif
+
+if get_option('use_gpl')
+  rz_analysis_sources += [
+    'p/analysis_arc.c',
+    'p/analysis_arm_gnu.c',
+    'p/analysis_cris.c',
+    'p/analysis_mips_gnu.c',
+    'p/analysis_nios2.c',
+    'p/analysis_ppc_gnu.c',
```
Comment (XVilka) [inline librz/analysis/meson.build]: Just remove PPC GNU completely, it seems also unmaintained and untested for years.

File: librz/analysis/meson.build [hunk 2]
```diff
@@ -165,6 +148,31 @@ rz_analysis_sources = [
...
 ]
 
+if capstone_dep.version().split('.')[0].to_int() > 4
+  rz_analysis_sources += 'p/analysis_riscv_cs.c'
+endif
+
+if get_option('use_gpl')
+  rz_analysis_sources += [
+    'p/analysis_arc.c',
+    'p/analysis_arm_gnu.c',
```
Comment (XVilka) [inline librz/analysis/meson.build]: Would be unnecessary after https://github.com/rizinorg/rizin/pull/2085

File: librz/asm/meson.build [hunk 3]
```diff
@@ -186,23 +140,78 @@ rz_asm_sources = [
...
+    'arch/arc/gnu/arcompact-dis.c',
+    'arch/arm/aarch64/aarch64-dis-2.c',
+    'arch/arm/aarch64/aarch64-dis.c',
+    'arch/arm/aarch64/aarch64-opc-2.c',
+    'arch/arm/aarch64/aarch64-opc.c',
+    'arch/arm/gnu/arm-dis.c',
+    'arch/arm/gnu/floatformat.c',
+    'arch/cris/gnu/cris-dis.c',
+    'arch/cris/gnu/cris-opc.c',
+    #'arch/hexagon/gnu/hexagon-dis.c',
```
Comment (XVilka) [inline librz/asm/meson.build]: Remove those - they were removed long time ago

File: librz/asm/meson.build [hunk 4]
```diff
@@ -186,23 +140,78 @@ rz_asm_sources = [
...
+    #'arch/hexagon/gnu/safe-ctype.c',
+    'arch/hppa/gnu/hppa-dis.c',
+    'arch/lanai/gnu/lanai-dis.c',
+    'arch/lanai/gnu/lanai-opc.c',
+    'arch/mips/gnu/mips-dis.c',
+    'arch/mips/gnu/mips-opc.c',
+    'arch/mips/gnu/mips16-opc.c',
+    'arch/nios/gnu/nios2-dis.c',
+    'arch/nios/gnu/nios2-opc.c',
+    'arch/ppc/gnu/ppc-dis.c',
```
Comment (XVilka) [inline librz/asm/meson.build]: Same, if you remove PPC GNU plugin.

File: librz/asm/meson.build [hunk 5]
```diff
@@ -186,23 +140,78 @@ rz_asm_sources = [
...
+    'arch/xtensa/gnu/elf32-xtensa.c',
+    'arch/xtensa/gnu/xtensa-dis.c',
+    'arch/xtensa/gnu/xtensa-isa.c',
+    'arch/xtensa/gnu/xtensa-modules.c',
+    #'arch/z80/expressions.c',
+    #'arch/z80/z80asm.c',
+    'p/asm_arc.c',
+    'p/asm_arm_gnu.c',
+    'p/asm_cris_gnu.c',
+    #'p/asm_hexagon_gnu.c',
```
Comment (XVilka) [inline librz/asm/meson.build]: Remove it - it's no longer in the tree for a looong time.

File: librz/meson.build [hunk 6]
```diff
@@ -322,9 +288,53 @@ demangler_plugins = [
...
+    'tricore',
+    'vax',
+    'xtensa',
+    'z80',
+  ]
+  asm_plugins += [
+    'arc',
+    'arm_gnu',
+    'cris_gnu',
+    #'hexagon_gnu',
```
Comment (XVilka) [inline librz/meson.build]: Remove this

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/analysis/meson.build (hunk 1 above)
Comment (XVilka) [reply librz/analysis/meson.build]: Could be done in a separate PR, like I did for ARM GNU and WineDbg

**Review by ret2libc — COMMENTED** [review]
LGTM!

**Review by XVilka — APPROVED** [review]

**Pull request merged**